### DISCUSSION TITLE
Allow to add an extra context to `handleSubmit` function

### DISCUSSION THREAD

**Discussion by mynavitechtus-sontd** [react-hook-form/react-hook-form#12419] [Q&A]
**Is your feature request related to a problem? Please describe.**

There is a case that users want to create validation schemes with dynamic params when validating. 
Check out this ticket for further information: https://github.com/jquense/yup/issues/337

```ts
// file: schema.js
// create the schema once
yup = require('yup');
const schema = yup.mixed().test(
  'my-test-name',
  'cannot be an existing value',
  function test(value) {
    return !this.options.context.myArray.includes(value)
  } 
);
module.exports = schema;


// other file:
schema = require('./schema.js');
let myArray = ['blue', 'green'];
schema.validateSync('yellow', { context: { myArray: myArray });
```

**Describe the solution you'd like**

```tsx
const myArray = useState([ ... ]);
// ...

<form onSubmit={handleSubmit(onSubmit, { context: { myArray } })}>
```

**Describe alternatives you've considered**

**Additional context**

**Comment by BrendanC23** [comment]
I haven't used Yup before, but it looks like this is already supported. See [the docs](https://react-hook-form.com/docs/useform#context) and the Yup resolver code.

https://github.com/react-hook-form/resolvers/blob/e523dde4d9f2ee0f6168fab3696ed4aeb59fc69a/yup/src/yup.ts#L64-L68

Also, see [this PR](https://github.com/react-hook-form/resolvers/pull/30).
